lib/ansible/modules/cloud/amazon/ec2_lc.py

```diff
@@ -225,79 +325,133 @@ def create_launch_config(connection, module):
+                bdm.update(create_block_device_meta(module, volume))
 
-    launch_configs = connection.get_all_launch_configurations(names=[name])
+    launch_configs = connection.describe_launch_configurations(
+        LaunchConfigurationNames=[name]).get('LaunchConfigurations')
     changed = False
-    if not launch_configs:
+    result = {}
+
+    launch_config = {
+        'LaunchConfigurationName': name,
+        'EbsOptimized': ebs_optimized,
+        'PlacementTenancy': placement_tenancy
+    }
+
+    if image_id is not None:
+        launch_config['ImageId'] = image_id
+
+    if instance_type is not None:
+        launch_config['InstanceType'] = instance_type
+
+    if instance_id is not None:
+        launch_config['InstanceId'] = instance_id
+
+    if classic_link_vpc_id is not None:
+        launch_config['ClassicLinkVPCId'] = classic_link_vpc_id
+
+    if instance_monitoring:
+        launch_config['InstanceMonitoring'] = {
+            'Enabled': advanced_instance_monitoring}
+
+    if placement_tenancy is not None:
+        launch_config['PlacementTenancy'] = placement_tenancy
+
+    if classic_link_vpc_security_groups is not None:
+        launch_config['ClassicLinkVPCSecurityGroups'] = (
+            classic_link_vpc_security_groups)
+
+    if key_name is not None:
+        launch_config['KeyName'] = key_name
+
+    if bdm is not None:
+        launch_config['BlockDeviceMappings'] = [bdm]
+
+    if security_groups is not None:
+        launch_config['SecurityGroups'] = security_groups
+
+    if kernel_id is not None:
+        launch_config['KernelId'] = kernel_id
+
+    if ramdisk_id is not None:
+        launch_config['RamdiskId'] = ramdisk_id
+
+    if instance_profile_name is not None:
+        launch_config['IamInstanceProfile'] = instance_profile_name
+
+    if spot_price is not None:
+        launch_config['SpotPrice'] = str(spot_price)
+
+    if assign_public_ip is not None:
+        launch_config['AssociatePublicIpAddress'] = assign_public_ip
+
+    if user_data is not None:
+        launch_config['UserData'] = user_data
+
+    if len(launch_configs) == 0:
         try:
-            connection.create_launch_configuration(lc)
-            launch_configs = connection.get_all_launch_configurations(names=[name])
+            connection.create_launch_configuration(**launch_config)
+            launch_configs = connection.describe_launch_configurations(
+                LaunchConfigurationNames=[name]).get('LaunchConfigurations')
             changed = True
-        except BotoServerError as e:
+        except botocore.exceptions.ClientError as e:
             module.fail_json(msg=str(e))
 
-    result = dict(
-                 ((a[0], a[1]) for a in vars(launch_configs[0]).items()
-                  if a[0] not in ('connection', 'created_time', 'instance_monitoring', 'block_device_mappings'))
+    if len(launch_configs) > 0:
+        result = dict(
+                     ((a[0], a[1]) for a in launch_configs[0].items()
```

Comment:
`dict((k, v) for k, v in launch_configs[0].items() if k not in ...)` is probably a bit more readable.